internal/rbd/driver/driver.go

```diff
@@ -201,7 +201,7 @@ func (r *Driver) Run(conf *util.Config) {
 
 	r.startProfiling(conf)
 
-	if conf.IsNodeServer {
+	if conf.IsNodeServer && os.Getenv("KUBERNETES_SERVICE_HOST") != "" {
```

Comment:
Let's make use of this function which we already have
https://github.com/ceph/ceph-csi/blob/2fe7dbe2af4e55bba69ef1e15e931426fe2c7a3a/internal/util/k8s/client.go#L64